### PR TITLE
add quarantine_size_mb

### PR DESCRIPTION
**Pull request by randyh62**

### CODE REVIEW

**Review by b-sumner — COMMENTED** [review]
Looks good to me.  I am not opposed to replacing `free`'d with freed.

**Comment by randyh62** [comment]
I will merge it and also cherry pick it for the other appropriate releases.

File: docs/conceptual/using-gpu-sanitizer.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ If it does not appear, when executed the application will quickly output an ASan
 
 There is an environment variable, `ASAN_OPTIONS`, that can be used to adjust the runtime behavior of the ASan runtime itself. There are more than a hundred "flags" that can be adjusted (see an old list at [flags](https://github.com/google/sanitizers/wiki/AddressSanitizerFlags)) but the default settings are correct and should be used in most cases. It must be noted that these options only affect the host ASan runtime. The device runtime only currently supports the default settings for the few relevant options.
 
-There are two `ASAN_OPTION` flags of particular note.
+There are three `ASAN_OPTION` flags of particular note.
```
Comment (lpaoletti) [inline docs/conceptual/using-gpu-sanitizer.md:95]:
```suggestion
There are three `ASAN_OPTION` flags of note.
```

File: docs/conceptual/using-gpu-sanitizer.md [hunk 2]
```diff
@@ -102,6 +102,13 @@ This tells the ASan runtime to halt the application immediately after detecting
...
 
+* `quarantine_size_mb=N default 256`
+
+This option defines the number of megabytes (MB) `N` of memory that the ASAN runtime will hold after it is `free`d to detect use-after-free situations.
+This memory is unavailable for other purposes. The default of 256 MB may be too small to detect some use-after-free situations, especially given that the large size of many GPU memory allocations may push `free`d allocations out of quarantine before the attempted use.
```
Comment (lpaoletti) [inline docs/conceptual/using-gpu-sanitizer.md:108]:
```suggestion
This memory is unavailable for other purposes. The default of 256 MB might be too small to detect some use-after-free situations, especially given that the large size of many GPU memory allocations might push `free`d allocations out of quarantine before the attempted use.
```

Comment (lpaoletti) [reply docs/conceptual/using-gpu-sanitizer.md:108]: i don't understand free'd

Does this make sense?
...many GPU memory allocations might push allocations that are now free out of quarantine before the attempted use.

File: docs/conceptual/using-gpu-sanitizer.md [hunk 3]
```diff
@@ -102,6 +102,13 @@ This tells the ASan runtime to halt the application immediately after detecting
 
 This option directs the ASan runtime to enable the [Leak Sanitizer](https://clang.llvm.org/docs/LeakSanitizer.html) (LSan). Unfortunately, for heterogeneous applications, this default will result in significant output from the leak sanitizer when the application exits due to allocations made by the language runtime which are not considered to be leaks. This output can be avoided by adding `detect_leaks=0` to the `ASAN_OPTIONS`, or alternatively by producing an LSan suppression file (syntax described [here](https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer)) and activating it with environment variable `LSAN_OPTIONS=suppressions=/path/to/suppression/file`. When using a suppression file, a suppression report is printed by default. The suppression report can be disabled by using the `LSAN_OPTIONS` flag `print_suppressions=0`.
 
+* `quarantine_size_mb=N default 256`
+
+This option defines the number of megabytes (MB) `N` of memory that the ASAN runtime will hold after it is `free`d to detect use-after-free situations.
```
Comment (lpaoletti) [inline docs/conceptual/using-gpu-sanitizer.md:107]: The ASAN occurrence here is all uppercase and not ASan

File: docs/conceptual/using-gpu-sanitizer.md [hunk 4]
```diff
@@ -102,6 +102,13 @@ This tells the ASan runtime to halt the application immediately after detecting
 
 This option directs the ASan runtime to enable the [Leak Sanitizer](https://clang.llvm.org/docs/LeakSanitizer.html) (LSan). Unfortunately, for heterogeneous applications, this default will result in significant output from the leak sanitizer when the application exits due to allocations made by the language runtime which are not considered to be leaks. This output can be avoided by adding `detect_leaks=0` to the `ASAN_OPTIONS`, or alternatively by producing an LSan suppression file (syntax described [here](https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer)) and activating it with environment variable `LSAN_OPTIONS=suppressions=/path/to/suppression/file`. When using a suppression file, a suppression report is printed by default. The suppression report can be disabled by using the `LSAN_OPTIONS` flag `print_suppressions=0`.
```
Comment (lpaoletti) [inline docs/conceptual/using-gpu-sanitizer.md:103]:
```suggestion
This option directs the ASan runtime to enable the [Leak Sanitizer](https://clang.llvm.org/docs/LeakSanitizer.html) (LSan). For heterogeneous applications, this default results in significant output from the leak sanitizer when the application exits due to allocations made by the language runtime which are not considered to be leaks. This output can be avoided by adding `detect_leaks=0` to the `ASAN_OPTIONS`, or alternatively by producing an LSan suppression file (syntax described [here](https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer)) and activating it with environment variable `LSAN_OPTIONS=suppressions=/path/to/suppression/file`. When using a suppression file, a suppression report is printed by default. The suppression report can be disabled by using the `LSAN_OPTIONS` flag `print_suppressions=0`.
```

File: docs/conceptual/using-gpu-sanitizer.md (hunk 2 above)
Comment (randyh62) [reply docs/conceptual/using-gpu-sanitizer.md:108]: I sent you a message regarding freed.

File: docs/conceptual/using-gpu-sanitizer.md (hunk 3 above)
Comment (randyh62) [reply docs/conceptual/using-gpu-sanitizer.md]: I made them consistent with ASan

File: docs/conceptual/using-gpu-sanitizer.md [hunk 5]
```diff
@@ -92,15 +92,22 @@ If it does not appear, when executed the application will quickly output an ASan
 
 There is an environment variable, `ASAN_OPTIONS`, that can be used to adjust the runtime behavior of the ASan runtime itself. There are more than a hundred "flags" that can be adjusted (see an old list at [flags](https://github.com/google/sanitizers/wiki/AddressSanitizerFlags)) but the default settings are correct and should be used in most cases. It must be noted that these options only affect the host ASan runtime. The device runtime only currently supports the default settings for the few relevant options.
 
-There are two `ASAN_OPTION` flags of particular note.
+There are three `ASAN_OPTION` flags of particular note.
```
Comment (randyh62) [inline docs/conceptual/using-gpu-sanitizer.md:95]: There are three `ASAN_OPTION` flags of note.

**Review by lpaoletti — COMMENTED** [review]
not related to this work but the three notes on this page aren't using the standard formatting for ntoes

image

**Review by lpaoletti — COMMENTED** [review]
The highlighted topic looks like it's part of Runtime Reporting. 
Should it be brought up a level to be parallel. 
Seems like this is related to overall and not just runtime.

**Review by lpaoletti — APPROVED** [review]

**Pull request merged**